03_Modules/Locations/src/module/admin.ts

```diff
@@ -0,0 +1,172 @@
+import { Service } from 'typedi';
+import { type ILogObj, Logger } from 'tslog';
+import {
+  ChargingStation,
+  Location,
+  SequelizeLocationRepository,
+} from '@citrineos/data';
+import {
+  AdminConnectorDTO,
+  AdminEVSEDTO,
+  AdminLocationDTO,
+  CitrineOcpiLocationMapper,
+  LocationDTO,
+  LocationsBroadcaster,
+  OcpiLocation,
+  OcpiLocationProps,
+  OcpiLocationRepository,
+  OcpiEvse,
+  OcpiEvseRepository,
+  OcpiConnector,
+  OcpiConnectorRepository,
+  VariableAttributesUtil,
+  OcpiLocationsUtil,
+  InvalidParamException,
+} from '@citrineos/ocpi-base';
+
+@Service()
+export class AdminLocationsService {
+  constructor(
+    private logger: Logger<ILogObj>,
+    private locationMapper: CitrineOcpiLocationMapper,
+    private locationRepository: SequelizeLocationRepository,
+    private ocpiLocationRepository: OcpiLocationRepository,
+    private ocpiEvseRepository: OcpiEvseRepository,
+    private ocpiConnectorRepository: OcpiConnectorRepository,
+    private locationsBroadcaster: LocationsBroadcaster,
+    private variableAttributesUtil: VariableAttributesUtil,
+    private ocpiLocationsUtil: OcpiLocationsUtil,
+  ) {}
+
+  public async createOrUpdateLocation(
+    adminLocationDto: AdminLocationDTO,
+    broadcast: boolean,
+  ): Promise<LocationDTO> {
+    this.logger.debug(
+      adminLocationDto.id
+        ? `Updating Location ${adminLocationDto.id}`
+        : `Creating Location ${adminLocationDto.name}`,
+    );
+
+    const [invalidLocation, message] = AdminLocationDTO.IS_LOCATION_INVALID(adminLocationDto);
```

Comment:
Should `IS_LOCATION_INVALID` be in standard camel case since it is more of a method?

Also, why not make this a non static method? That way we can write:
`adminLocationDto.isValid()`?

Reply:
problem solved now that this method is removed thanks to the built in validation methods from our class validation annotations 😎